### DISCUSSION TITLE
What is our final project like for the class?

### DISCUSSION THREAD

**Discussion by coopsand** [USU-ITLS-x265-23-summer/class-repo#10] [Q&A] [answered]
I wanted to ask the following question for Task 2 of Week 1: What is our final project like in the class? Do we get to walk away with a working website that we can use in our portfolio?

-Cooper Sanders (A023324659)
"coopsand"

**Comment by metasean** [comment, marked as answer]
>Do we get to walk away with a working website that we can use in our portfolio?

tldr:  Ideally, but ultimately it's up to you.  

https://github.com/USU-ITLS-x265-23-summer/class-repo/blob/main/Weekly-assignments/Projects.md

---

If you work through all the weekly tasks, and set a reasonable project goal, then you should absolutely have the basic skills to make a working website in your portfolio that you're proud of.  (Many of you won't feel like you have the knowledge.  "Imposter syndrome" is a real thing, and many developers fight it, even when they reach senior levels.  But, the many, many tasks that you'll work through will give you enough knowledge and practice that you absolutely will have the skill, even if you lack the confidence!) 

Starting next week, we'll begin to set expectations for _your_ projects, which will largely be driven by what you're learning through the weekly exercises and how much effort you want to put into it.  (e.g. If you're struggling with JavaScript, you'll probably want to settle on a project that uses less JavaScript than other students.  However, if you're more focused on providing a lot of interactions, then you'll probably lean toward a project that's heavy in JavaScript, with less focus on CSS.)

One of many reasons that I'm emphasizing more discussion, is that there were two students last summer, who absolutely had the skill to do more with their projects, but in both cases didn't have a lot of confidence in themselves and they were embarrassed to ask questions.  In both cases, they had simple typos (which every single developer has to deal with, and sometimes they can be hard to find).  Instead of asking for help, they made progress in other areas, but spent several weeks going in circles on one area because of typos.  In both cases, during their weekly updates they showed progress in other areas, and avoided mention of the problems they were having elsewhere.  They didn't ask for help until the very last week.  In both cases, they still had something that they could put in their portfolios, but is wasn't at the level that they were proud of.  It's genuinely heartbreaking to know that someone could have done so much more, if only they had asked for help sooner.  So when I harp on asking questions, and answering other people, and in the near future I'll start harping on posting your buggy code and helping others with their buggy code it's because I don't want you to just have something that you can put in your portfolio, but I want you to be able to put something in that you're genuinely proud to have in your portfolio.

**Comment by CJ3315** [comment]
My question is what is the best way to navigate the material in the class? Where should we be spending the bulk of our time? Should we treat Github as our canvas course? (This is new to me so I am just wondering what approach we should take to the learning structure)

**Reply by metasean** [reply]
Different question and conversation (i.e. not particularly related to "What is our final project like for the class?") - converted into its own [discussion #26](https://github.com/orgs/USU-ITLS-x265-23-summer/discussions/16).
